src/core/MOM.F90

```diff
@@ -2412,12 +2444,29 @@ subroutine initialize_MOM(Time, Time_init, param_file, dirs, CS, restart_CSp, &
       deallocate(T_in)
       deallocate(S_in)
     endif
+    if (use_ice_shelf) &
+      deallocate(frac_shelf_in)
   else
-    call MOM_initialize_state(CS%u, CS%v, CS%h, CS%tv, Time, G, GV, US, &
-        param_file, dirs, restart_CSp, CS%ALE_CSp, CS%tracer_Reg, &
-        CS%sponge_CSp, CS%ALE_sponge_CSp, CS%OBC, Time_in)
+    if (use_ice_shelf) then
+      allocate(CS%frac_shelf_h(isd:ied, jsd:jed))
+      CS%frac_shelf_h(:,:) = 0.0
+      call ice_shelf_query(ice_shelf_CSp,G,CS%frac_shelf_h)
+
+      call MOM_initialize_state(CS%u, CS%v, CS%h, CS%tv, Time, G, GV, US, &
+          param_file, dirs, restart_CSp, CS%ALE_CSp, CS%tracer_Reg, &
+          CS%sponge_CSp, CS%ALE_sponge_CSp, CS%OBC, Time_in, &
+          frac_shelf_h=CS%frac_shelf_h)
+    else
+      call MOM_initialize_state(CS%u, CS%v, CS%h, CS%tv, Time, G, GV, US, &
+          param_file, dirs, restart_CSp, CS%ALE_CSp, CS%tracer_Reg, &
+          CS%sponge_CSp, CS%ALE_sponge_CSp, CS%OBC, Time_in)
+    endif
   endif
 
+  if (use_ice_shelf) &
+    call hchksum(CS%frac_shelf_h, "MOM:frac_shelf_h", G%HI, &
+        haloshift=0, scale=GV%H_to_m)
```

Comment:
`GV%H_to_m` is not the right scaling factor for what I believe to be a nondimensional variable. Either the scale argument to this `hchksum()` call is wrong or the documentation describing the units of CS%frac_shelf_h is wrong. Regardless of which is the case, they need to be made consistent and correct.

Reply:
The scale factor is wrong. I am not sure why this is passing the rescaling tests. @marshallward ?

Reply:
I assume `use_ice_shelf=.true.` ?

Reply:
Yes

Reply:
This will be fixed after #1220 is merged.  It was requested that I submit those fixes in a separate PR.